internal/services/nginx/nginx_deployment_resource.go

```diff
@@ -627,6 +690,52 @@ func (m DeploymentResource) Read() sdk.ResourceFunc {
 						output.UpgradeChannel = props.AutoUpgradeProfile.UpgradeChannel
 					}
 
+					if props.NginxAppProtect != nil {
+						if props.NginxAppProtect.WebApplicationFirewallSettings.ActivationState != nil {
+							output.WebApplicationFirewallSettings = []WebApplicationFirewallSettings{
+								{
+									string(*props.NginxAppProtect.WebApplicationFirewallSettings.ActivationState),
+								},
+							}
+						}
+						if props.NginxAppProtect.WebApplicationFirewallStatus != nil {
+							wafStatus := WebApplicationFirewallStatus{}
+							if props.NginxAppProtect.WebApplicationFirewallStatus.AttackSignaturesPackage != nil {
```

Comment:
we can clean this up by
```suggestion
					if app := props.NginxAppProtect, apps != nil {
						if state := app.WebApplicationFirewallSettings.ActivationState; state != nil {
							output.WebApplicationFirewallSettings = []WebApplicationFirewallSettings{
								{
									string(*state),
								},
							}
						}
						if status := app.WebApplicationFirewallStatus; status != nil {
							wafStatus := WebApplicationFirewallStatus{}
							if pacakge := status.AttackSignaturesPackage; package  != nil {
```
etc, it would make this much easier to read/review.

also i think WebApplicationFirewallStatus needs to be nil checked?

Reply:
there is already a nil check for WebApplicationFirewallStatus. If you meant WebApplicationFirewallSettings, its not a pointer so excluded the nil check